esmvalcore/preprocessor/_time.py

```diff
@@ -161,6 +161,33 @@ def get_time_weights(cube):
     time_weights = time_thickness * ones
     return time_weights
 
+def time_statistics(cube, operator):
```

Comment:
I think it would be better to implement this function so it accepts a list of operators instead of just one.

Reply:
By the kind of operators we are using, this will mean producing several cubes. Can we mange that at this preprocessor step? I will very glad to have the option to get seasonal mean and median at the same time but I don know if the preprocessor can handle that

Reply:
Yes, it works fine, this is also an option for the mulitmodel_statistics preprocessor function

Reply:
Great! I will check the multimodel module and implement it following that guide

Reply:
Thinking a bit more about it, it may not be trivial to get this working correctly with the provenance recording..

Reply:
Yes, I felt a bit lost while looking to the code. Maybe we can keep this merge small(er) by making it to support only one operator and then add the others. 

I also added other things related to time operations that I need to simplify seaice drift and other metrics, and don't want to keep them waiting for this fix